harness/determined/pytorch/_pytorch_context.py

```diff
@@ -456,10 +456,11 @@ def configure_apex_amp(
             If  ``optimizers`` args were lists, the corresponding return value will
             also be a list.
         """
-        if not self.env.managed_training:
+        if not enabled or not self.env.managed_training:
             return models, optimizers
 
-        check.is_none(self._scaler, "Do not mix APEX with PyTorch AMP")
+        if self._scaler is not None:
+            check.false(self._scaler.is_enabled(), "Do not mix APEX with PyTorch AMP")
```

Comment:
The `check` module is legacy code.  Any time you touch it, please just replace it with an if statement.

Basically it is going to give low-quality error messages ("expected false but got true" or something), and it's going to raise a meaningless error type.

So please make this an if statement raising an exception of some sort.  Maybe InvalidExperimentException or whatever we call it.

Reply:
I'm glad to know that. I didn't like `check`. :-p